server/current-context.js

```diff
@@ -78,7 +78,28 @@ LoopBackContext.createContext = function(scopeName) {
     process.context[scopeName] = ns;
     // Set up LoopBackContext.getCurrentContext()
     LoopBackContext.getCurrentContext = function() {
-      return ns && ns.active ? ns : null;
+      if (ns && ns.active) {
+        /**
+         * **NOTE**
+         * This only re-binds get and set methods, the most used.
+         * If you use other methods of the context, e.g. runInContext(), etc.,
+         * you may run into unexpected issues that are fixed only for get, set.
+         */
+        var boundMethods = {
+          get: ns.bind(ns.get),
+          set: ns.bind(ns.set),
+        };
+        var handler = {
+          get: function(target, name) {
+            return ['get', 'set'].includes(name) ?
+            boundMethods[name] :
+            target[name];
+          },
+        };
+        return new Proxy(ns, handler);
```

Comment:
`Proxy` also seems to be not available in Node 4.x, see https://travis-ci.org/strongloop/loopback-context/jobs/189233795#L264

```
Unhandled error for request GET /TestModels/test:
ReferenceError: Proxy is not defined
    at Object.LoopBackContext.getCurrentContext (/home/travis/build/strongloop/loopback-context/server/current-context.js:99:20)
```